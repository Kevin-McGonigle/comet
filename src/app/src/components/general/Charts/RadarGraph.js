import React from 'react';
import { Radar } from 'react-chartjs-2';

const RadarGraph = (props) => {
    // const { data } = props;
    const data = {
        labels: ['Cyclomatic Complexity', 'Logical Lines of Code', 'Lines of Comment', 'Afferent Coupling', 'Efferent Coupling', 'Instability', 'Method Cohesion', 'Relational Cohesion', 'Nesting Depth'],
        datasets: [
            {
                label: 'General Overview',
                data: [10, 20, 30, 40, 50, 35, 60, 20, 20],
                backgroundColor: "rgba(200,0,0,0.2)",
                title: {
                    display: true,
                    text: "Radar Graph"
                }
            }
        ]
    };

    return (
            <Radar 
                data={data}
                width={150}
                height={150}
                options={{ maintainAspectRatio: false, backgroundColor: "white" }}
            />
    )
}

export default RadarGraph;