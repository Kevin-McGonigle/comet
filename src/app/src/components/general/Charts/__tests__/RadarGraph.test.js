import React from 'react';
import RadarGraph from '../RadarGraph';
import { shallow } from 'enzyme';
import "../../../../setupTests"

test('should render RadarGraph as expected', () => { 
    const component = shallow(
        <RadarGraph />
    );
    expect(component.html()).toMatchSnapshot();
});
