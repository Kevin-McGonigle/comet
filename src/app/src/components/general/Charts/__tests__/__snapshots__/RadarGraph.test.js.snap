// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render RadarGraph as expected 1`] = `"<canvas height=\\"150\\" width=\\"150\\"></canvas>"`;
